src/jit/regalloc.cpp

```diff
@@ -335,17 +335,14 @@ void Compiler::raMarkStkVars()
             /*
               For Debug Code, we have to reserve space even if the variable is never
               in scope. We will also need to initialize it if it is a GC var.
-              So we set lvMustInit and artifically bump up the ref-cnt.
+              So we set lvMustInit and verify it has a nonzero ref-cnt.
              */
 
             if (opts.compDbgCode && !stkFixedArgInVarArgs && lclNum < info.compLocalsCount)
             {
-                needSlot |= true;
+                assert(varDsc->lvRefCnt() > 0);
```

Comment:
Would it make sense here to set it to one if it is, in fact zero, to avoid having to make this a `noway_assert`?

Reply:
I suppose it can't hurt -- it should be quite difficult now to create unreferenced locals in debug or minopts but this seems like cheap insurance.